src/Sulu/Bundle/SearchBundle/Resources/public/js/components/search-results/main.js

```diff
@@ -0,0 +1,435 @@
+                    if (!this.categoriesStore[key]) {
+                        this.categoriesStore[key] = data[key];
+                    } else {
+                        this.categoriesStore[key].results = this.categoriesStore[key].results.concat(data[key].results);
+                    }
+                }.bind(this));
+            }
+
+            return this.categoriesStore;
+        },
+
+        /**
+         * @method dropDownInputActionHandler
+         */
+        dropDownInputActionHandler: function(data) {
+            if (!!data.value) {
+                this.state.query = data.value;
+                this.state.category = data.selectedElement;
+                this.state.page = 1;
+                this.categoriesStore = {};
+
+                this.startLoader();
+                this.updateResults();
+                this.load().then(function(data) {
+                    this.categoriesStore = data;
+                    this.updateResults(data);
+                }.bind(this));
+            }
+        },
+
+        /**
+         * @method dropDownInputClearHandler
+         */
+        dropDownInputClearHandler: function() {
+            this.updateResults();
+        },
+
+        /**
+         * @method getEntryDeepUrl
+         * @param {String} category
+         * @param {Object} data
+         */
+        getEntryDeepUrl: function(category, data) {
+            var handler = this.urlTemplateMapping[category],
+                deepUrl = null;
+
+            if (handler) {
+                deepUrl = handler.call(this, data);
+            }
+
+            return deepUrl;
+        },
+
+        /**
+         * @type {Object}
+         */
+        urlTemplateMapping: {
+            page: function(data) {
+                if (data.url === '/') {
+                    // startpage
+                    return this.sandbox.urlManager.getUrl('startpage', {
+                        id: data.id,
+                        webspace: data.properties.webspace_key
+                    });
+                } else {
+                    return this.sandbox.urlManager.getUrl('contentDetail', {
+                        id: data.id,
+                        webspace: data.properties.webspace_key
+                    });
+                }
+            },
+
+            contact: function(data) {
+                return this.sandbox.urlManager.getUrl('contactDetail', data);
+            },
+
+            account: function(data) {
+                return this.sandbox.urlManager.getUrl('accountDetail', data);
+            },
+
+            media: function(data) {
+                return this.sandbox.urlManager.getUrl('mediaDetail', {
+                    id: data.id,
+                    collectionId: data.properties.collection_id
+                });
+            },
+
+            snippet: function(data) {
+                return this.sandbox.urlManager.getUrl('snippetDetail', data);
+            }
```

Comment:
This information should somehow be offered by the corresponding bundles instead of centralized in this place.